st2api/st2api/validation.py

```diff
@@ -15,10 +15,58 @@
 
 from oslo_config import cfg
 
-__all__ = ["validate_rbac_is_correctly_configured"]
+from webob import cookies
 
+__all__ = [
+    "validate_auth_cookie_is_correctly_configured",
+    "validate_rbac_is_correctly_configured",
+]
 
-def validate_rbac_is_correctly_configured():
+
+def validate_auth_cookie_is_correctly_configured() -> bool:
+    """
+    Function which verifies that SameCookie config option value is correctly configured.
+
+    This method should be called in the api init phase so we catch any misconfiguration issues
+    before startup.
+    """
+    if cfg.CONF.api.auth_cookie_same_site not in ["strict", "lax", "none", "None"]:
+        raise ValueError(
+            'Got invalid value "%s" (type %s) for cfg.CONF.api.auth_cookie_same_site config '
+            "option. Valid values are: strict, lax, none, None."
+            % (
+                cfg.CONF.api.auth_cookie_same_site,
+                type(cfg.CONF.api.auth_cookie_same_site),
+            )
+        )
+
+    # Now we try to make a dummy cookie to verify all the options are configured correctly. Some
+    # Options are mutually exclusive - e.g. SameSite none and Secure false.
+    try:
+        # NOTE: None and none don't mean the same thing - None implies not setting this attribute
+        # (backward compatibility) and none implies setting this attribute value to none
```

Comment:
wow. That's a very minute distinction - could we use "unset" instead of "None"?

Reply:
Yeah, that's how it's handled by the browser, I'm not a big fan of it either.

I'm fine for using a different more explicit value such as ``unset``.